ykcompile/src/lib.rs

```diff
@@ -1115,4 +1225,58 @@ mod tests {
         ct.execute(&mut args2);
         assert_eq!(args2.0, 39);
     }
+
+    #[inline(never)]
+    fn checked_add(a: u8) -> u8 {
```

Comment:
This is only a `checked_add` when debug assertions are enabled. The integer types have a `checked_add` method that is guaranteed to be a checked add. It uses the `add_with_overflow` intrinsic internally.

Reply:
Tracing currently only works with debug enabled, but this is useful to keep in mind, if/when we managed to get tracing to work without debug mode.

Reply:
There is a differemce between debuginfo and debug assertions. The former is -Cdebuginfo the later -Cdebug-assertions. They can independently be changed.

Reply:
Fixed in 9b3d3f5. I've renamed the functions to make clear that these tests can also be used to test unchcked `add` (e.g. by passing `overflow-checks=off`). However, doing so currently results in an `UnimplementedError`, as we haven't implemented unchecked `add` yet.